kedro/contrib/io/matplotlib/matplotlib_s3_writer.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2018-2019 QuantumBlack Visual Analytics Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, AND
+# NONINFRINGEMENT. IN NO EVENT WILL THE LICENSOR OR OTHER CONTRIBUTORS
+# BE LIABLE FOR ANY CLAIM, DAMAGES, OR OTHER LIABILITY, WHETHER IN AN
+# ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF, OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+#
+# The QuantumBlack Visual Analytics Limited ("QuantumBlack") name and logo
+# (either separately or in combination, "QuantumBlack Trademarks") are
+# trademarks of QuantumBlack. The License does not grant you any right or
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""
+``MatplotlibWriterS3`` saves matplotlib objects as image files to S3.
+"""
+
+import copy
+import io
+from typing import Any, Dict, List, Optional, Union
+
+from matplotlib.pyplot import figure
+from s3fs import S3FileSystem
+
+from kedro.io import AbstractDataSet, DataSetError
+
+
+class MatplotlibWriterS3(AbstractDataSet):
+    # pylint: disable=too-many-instance-attributes
+    """
+    ``MatplotlibWriter`` saves matplotlib objects as image files.
+
+    Example:
+    ::
+
```

Comment:
Is it necessary to add the `>>>`? I prefer without because it means I can copy paste the example and it will work. By adding the `>>>`, I always have to copy paste, and delete `>>>` from every line.

Reply:
I understand, I always get annoyed by that actually. `>>>` is more for the docs (see example code. https://kedro.readthedocs.io/en/latest/kedro.io.CSVLocalDataSet.html#kedro.io.CSVLocalDataSet)

Reply:
done!